src/pages/studyView/customCaseSelection/CustomCaseSelection.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Sample } from 'shared/api/generated/CBioPortalAPIInternal';
+import { observer } from "mobx-react";
+import { bind } from 'bind-decorator';
+import Draggable from 'react-draggable';
+import classnames from 'classnames';
+import styles from "./styles.module.scss";
+import { FlexRow } from 'shared/components/flexbox/FlexBox';
+import { observable, computed, action } from 'mobx';
+import { ButtonGroup, Radio } from 'react-bootstrap';
+
+export interface ICustomCaseSelectionProps {
+    selectedSamples: Sample[];
+    onClose: () => void;
+    onSubmit:(samples:Sample[]) => void;
+}
+
+const GroupByOptions: { value: 'sample'|'patient'; label: string;}[] = [
+    {value: 'sample', label: 'By sample ID'},
+    {value: 'patient', label: 'By patient ID'}
+];
+
+@observer
+export default class CustomCaseSelection extends React.Component<ICustomCaseSelectionProps, {}> {
+
+    @observable showCaseIds: boolean = false;
+    @observable caseIdsMode: 'sample' | 'patient' = 'sample';
+    @observable caseIds: string = ''
+    @observable invalidCases: string[] = [];
+    @observable validCases: Sample[] = []
+
+    @computed get sampleSet(): { [id: string]: Sample } {
+        return _.reduce(this.props.selectedSamples, (acc: { [id: string]: Sample }, next) => {
```

Comment:
you can write this as `_.keyBy(this.props.selectedSamples, s=>`${s.studyId}:${s.sampleId}`)`

Reply:
User actually enters `sampleId` or `studyId:sampleId` in text area. 
Is there a login in front-end to convert `studyId:sampleId` to `uniqueSampleKey`?

Reply:
oh I see, this seems fine and a little different than whats going on in Query

Reply:
It should be the same though. Just curious how this different from the one in Query

Reply:
Query page includes some logic if Virtual studies are in selection. eventually this component should be updated to support that functionality and replace it(what's in query page)

Reply:
the difference I was thinking of is that here theres already the selected study, and the samples in the study are already known, so you dont need to fetch more. that may be wrong though and im misunderstanding whats going on here

Reply:
yes, even that's true.